src/mon/PGMonitor.cc

```diff
@@ -699,6 +699,11 @@ bool PGMonitor::preprocess_pg_stats(MonOpRequestRef op)
     return true;
   }
 
+  if (stats->fsid != mon->monmap->fsid) {
+    dout(0) << "preprocess_pg_stats on fsid " << stats->fsid << " != " << mon->monmap->fsid << dendl;
```

Comment:
it would be nice to also print out the message that triggered this. Say,

`dout(0) << __func__ << " drop message on fsid " << stats->fsid << " != " << mon->monmap->fsid << " for " << *stats << dendl;`

or something. Same for the other parts where you add this check.